### PR TITLE
Adding a check that ensures the app required for the database backend…

### PR DESCRIPTION
**Pull request by dennisstritzke**
… is installed.

Although the proper setup is documented in the README, it is easy to miss and (a little) hassle to debug without knowledge of the django_tasks code itself.

### CODE REVIEW

**Comment by dennisstritzke** [comment]
An improper configuration would cause the following check error
```console
$ pipenv run python manage.py check
SystemCheckError: System check identified some issues:

ERRORS:
?: DatabaseBackend configured as django_tasks backend, but database app not installed
	HINT: Insert 'django_tasks.backends.database' after 'django_tasks' in INSTALLED_APPS
```

File: django_tasks/backends/database/backend.py [hunk 1]
```diff
@@ -88,3 +91,18 @@ async def aget_result(self, result_id: str) -> TaskResult:
...
+    def check(self, **kwargs: Any) -> List[CheckMessage]:
+        if "django_tasks.backends.database" not in settings.INSTALLED_APPS:
+            backend_name = self.__class__.__name__
+            package_name = sys.modules[__name__].__package__
+
+            return [
+                CheckMessage(
+                    ERROR,
+                    f"{backend_name} configured as django_tasks backend, but database app not installed",
+                    f"Insert '{package_name}' after 'django_tasks' in INSTALLED_APPS",
```
Comment (RealOrangeOne) [inline django_tasks/backends/database/backend.py:104]: Question: Does it _have_ to be after?

Nit: You're not currently checking that it's installed after.

Comment (dennisstritzke) [reply django_tasks/backends/database/backend.py]: It does *not* have to come after. I removed that part of the instruction.

File: django_tasks/backends/database/backend.py [hunk 2]
```diff
@@ -88,3 +91,18 @@ async def aget_result(self, result_id: str) -> TaskResult:
             return (await DBTaskResult.objects.aget(id=result_id)).task_result
         except (DBTaskResult.DoesNotExist, ValidationError) as e:
             raise ResultDoesNotExist(result_id) from e
+
+    def check(self, **kwargs: Any) -> List[CheckMessage]:
+        if "django_tasks.backends.database" not in settings.INSTALLED_APPS:
+            backend_name = self.__class__.__name__
+            package_name = sys.modules[__name__].__package__
```
Comment (RealOrangeOne) [inline django_tasks/backends/database/backend.py:98]: Given we're hardcoding the module name above, we should just hardcode it here too.

File: django_tasks/backends/database/backend.py [hunk 3]
```diff
@@ -88,3 +91,18 @@ async def aget_result(self, result_id: str) -> TaskResult:
             return (await DBTaskResult.objects.aget(id=result_id)).task_result
         except (DBTaskResult.DoesNotExist, ValidationError) as e:
             raise ResultDoesNotExist(result_id) from e
+
+    def check(self, **kwargs: Any) -> List[CheckMessage]:
+        if "django_tasks.backends.database" not in settings.INSTALLED_APPS:
```
Comment (RealOrangeOne) [inline django_tasks/backends/database/backend.py:96]: Suggestion:

```python
from django.apps import apps

if apps.is_installed("django_tasks.backends.database"):
```

This uses Django's actual APIs for checking if an app is installed

Comment (dennisstritzke) [reply django_tasks/backends/database/backend.py]: That is way better. Thank you for pointing to that API

**Pull request merged**